source/PreferencesPanel.cpp

```diff
@@ -876,10 +876,58 @@ void PreferencesPanel::DrawPlugins()
 				top.Y() += sprite->Height() + 10.;
 			}
 
+			string text;
+			if(!plugin.version.empty())
+			{
+				text += "Version: " + plugin.version + '\n';
+			}
+			if(!plugin.authors.empty())
+			{
+				text += "Authors: ";
+				for(const string &author : plugin.authors)
+					text += author + ',';
+				text.pop_back();
+				text += '\n';
+			}
+			if(!plugin.tags.empty())
+			{
+				text += "Tags: ";
+				for(const string &tag : plugin.tags)
+					text += tag + ',';
+				text.pop_back();
+				text += '\n';
+			}
+			if(!plugin.dependencies.IsEmpty())
+			{
+				text += "Dependencies:\n";
+				if(!plugin.dependencies.gameVersion.empty())
+					text += "  Game Version: " + plugin.dependencies.gameVersion + '\n';
+				if(!plugin.dependencies.required.empty())
+				{
+					text += "  Requires:\n";
+					for(const string &dependency : plugin.dependencies.required)
+						text += "  - " + dependency + '\n';
+				}
+				if(!plugin.dependencies.optional.empty())
+				{
+					text += "  Optional:\n";
+					for(const string &dependency : plugin.dependencies.optional)
+						text += "  - " + dependency + '\n';
+				}
+				if(!plugin.dependencies.conflicted.empty())
+				{
+					text += "  Conficts:\n";
+					for(const string &dependency : plugin.dependencies.conflicted)
+						text += "  - " + dependency + '\n';
+				}
+				text += '\n';
+			}
```

Comment:
I feel like Plugin should have some function that constructs the text to be displayed in the description box instead of that being something that PreferencesPanel builds up. Thoughts? (It's honestly probably inconsequential, I just feel like it makes more sense for a Plugin to say "here's how I want to be displayed" instead of handing the Plugin to PreferencesPanel and having it ask "ok, how do I display this?")

Reply:
I'm a fan of this change, the way descriptions are constructed will almost certainly be changed again in the future so separating it should help a lot for maintainability.